src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -863,13 +870,23 @@ public static object DeepCopyInner(object original)
             {
                 copy = copier(original);
                 SerializationContext.Current.RecordObject(original, copy);
+                return copy;
             }
-            else
+
+            var externalSerializer = externalSerializers.FirstOrDefault(serializer => serializer.IsSupportedType(t));
```

Comment:
I'd recommend a slight change to this proposal to allow the same type to be serialized in different ways within the same method call, if this is the direction we're going.  For example:

```csharp
Task<bool> AuthenticateAsync(string user, string password);
```
in this scenario, we'd want the same types to have different serializers (the password being encrypted). The slight change would be to allow developers to override the serializer on a by-parameter basis.

```csharp
Task<bool> AuthenticateAsync(
    string user,
    [OrleansSerializer(typeof(EncryptionSerializer)]
    string password)
```
if no serializer is specified, the one from the configuration is used.  If the attribute is specified, then that one will always be used regardless of configuration changes.